lib/connection.js

```diff
@@ -83,14 +84,19 @@ connection.prototype.connect = function(callback){
   });
 };
 
-connection.prototype.end = function(){
+connection.prototype.end = function(callback){
+  callback = getCallback(callback);
   var self = this;
   self.connected = false;
   // Only disconnect if we established the redis connection on our own.
-  if (!self.options.redis && self.options.pkg !== 'fakeredis'){
-    if(typeof self.redis.disconnect === 'function'){ self.redis.disconnect(); }
-    else{ self.redis.quit(); }
+  if (!self.options.redis && self.options.pkg !== 'fakeredis') {
```

Comment:
@evantahler I placed the callback below this `if` block because neither `redis.disconnect` nor `redis.quit` accept a callback parameter. Really, this was just intended to allow for `promisifyAll()` to use it.